example/newreposecret/main.go

```diff
@@ -0,0 +1,197 @@
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"flag"
+	"fmt"
+	sodium "github.com/GoKillers/libsodium-go/cryptobox"
+	"github.com/google/go-github/v32/github"
+	"golang.org/x/oauth2"
+	"log"
+	"os"
+)
+
+var (
+	repo  = flag.String("repo", "", "The repo that the secret should be added to, ex. go-github")
+	owner = flag.String("owner", "", "The owner of there repo this should be added to, ex. google")
+)
+
+/*
+main creates a new secret in github for a given owner/repo based on a secretName passed in as an arg
+and a github token, and secret value provided via the environment
+
+usage:
+	export GITHUB_AUTH_TOKEN=<auth token from github that has secret create rights>
+	export SECRET_VARIABLE=<secret value of the secret variable>
+	go run main.go -owner <owner name> -repo <repository name> SECRET_VARIABLE
+
+ex:
+	export GITHUB_AUTH_TOKEN=0000000000000000
+	export SECRET_VARIABLE="my-secret"
+	go run main.go -owner google -repo go-github SECRET_VARIABLE
+
+Once it runs go to the github repository > settings > left side options bar > Secrets
+And you should see the new secret appear there.
+*/
+func main() {
+	flag.Parse()
+
+	token := os.Getenv("GITHUB_AUTH_TOKEN")
+	if token == "" {
+		log.Fatal(`Unauthorized: No token present in environment variables. 
+Add your github api token as an environment variable named GITHUB_AUTH_TOKEN
+
+ex:
+	export GITHUB_AUTH_TOKEN=<github auth token>`)
+	}
+
+	if *repo == "" {
+		log.Fatal(`required flag repo was not passed. 
+please pass the repo you want to add this flag to as part of the command
+
+ex:
+	go run main.go --repo <repo name> --owner <repo owner name> 
+`)
+	}
+
+	if *owner == "" {
+		log.Fatal(`required flag owner was not passed. 
+please pass the owner you want to add this flag to as part of the command
+
+ex:
+	go run main.go --repo <repo name> --owner <repo owner name> 
+`)
+	}
+
+	ctx, client, err := GithubAuth()
+	if err != nil {
+		log.Fatal("unable to authorize client with github using token found in env var GITHUB_AUTH_TOKEN, error was passed: ", err.Error())
+	}
+	secretName, err := getSecretName()
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	secretValue, err := getSecretValue(secretName)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	addedSecretName, err := AddRepoSecret(ctx, client, *owner, *repo, secretName, secretValue)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	fmt.Println(fmt.Sprintf("Added Secret %s to the repo %s/%s", addedSecretName, *owner, *repo))
+
+}
+
+func getSecretName() (string, error) {
+	if len(os.Args) < 2 {
+		err := errors.New("no secrets passed in the go run command. pass in a secret name and then ensure to have the secret variable as an environment variable")
+		return "", err
+	}
+	args := os.Args
+	secret := args[3:]
+	return secret[0], nil
```

Comment:
These last 3 lines are unnecessary, and since you are using the `flag` package, please don't use `os` as well... just use `flag.NArgs` and `flag.Arg` here.

Reply:
updated!